runtime/drivers/s3/model_manager.go

```diff
@@ -0,0 +1,113 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/bmatcuk/doublestar/v4"
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type ModelResultProperties struct {
+	Path string `mapstructure:"path"`
+}
+
+func (c *Connection) Rename(ctx context.Context, res *drivers.ModelResult, newName string, env *drivers.ModelEnv) (*drivers.ModelResult, error) {
+	return nil, nil
+}
+
+func (c *Connection) Exists(ctx context.Context, res *drivers.ModelResult) (bool, error) {
+	return true, nil
+}
+
+func (c *Connection) Delete(ctx context.Context, res *drivers.ModelResult) error {
+	cfg, err := c.awsConfig(ctx, "")
+	if err != nil {
+		return err
+	}
+	p := &ModelResultProperties{}
+	if err := mapstructure.Decode(res.Properties, p); err != nil {
+		return err
+	}
+	u, err := url.Parse(p.Path)
+	if err != nil {
+		return err
+	}
+	base, _ := doublestar.SplitPattern(strings.TrimPrefix(u.Path, "/"))
+	return deleteObjectsInPrefix(ctx, cfg, u.Host, base)
```

Comment:
Not sure, but can the `ModelResultProperties.Path` be a fixed path to a directory, file, or list of files (instead of a glob pattern)? Since glob patterns are for selecting files to ingest, not files to output. Also, by deleting the full prefix here it might delete files that actually don't match the full glob pattern, which seems weird (even if it doesn't matter in practice).

Reply:
If it is a fixed path to a directory then it is essentially the same as what we are doing here. 
To generate a list of files we will need to do a `ListObjects` call in `snowflake->object_store` connector. 

To ingest files using clickhouse's s3 integration we need a glob_pattern. Since a user can submit a s3->clickhouse model directly we need to make sure that the `input_properties` cover all cases.